summingbird-storm/src/main/scala/com/twitter/summingbird/storm/ComponentProvider.scala

```diff
@@ -0,0 +1,32 @@
+package com.twitter.summingbird.storm
+
+import com.twitter.algebird.Semigroup
+import com.twitter.summingbird.batch.Batcher
+import com.twitter.summingbird.online.executor.KeyValueShards
+import com.twitter.summingbird.storm.builder.Topology
+
+/**
+ * This trait describes how to build topology component which corresponds to `Producer`s chain.
+ */
+private[storm] trait ComponentProvider {
+  import StormTopologyBuilder._
+
+  /**
+   * Assuming this component corresponds to `Producer[Storm, T]` (which means it emits `Item[T]`)
```

Comment:
Something on the lines of:
Create a component that emits events with single type i.e. events that don't have to have a key-value structure. e.g. ...